vioscsi/vioscsi.c

```diff
@@ -500,13 +500,20 @@ ENTER_FN();
     if(adaptExt->dump_mode) {
         ConfigInfo->NumberOfPhysicalBreaks  = SCSI_MINIMUM_PHYSICAL_BREAKS;
     } else {
-        adaptExt->max_physical_breaks = MAX_PHYS_SEGMENTS;
+        adaptExt->max_physical_breaks = min(
+                                            max(SCSI_MINIMUM_PHYSICAL_BREAKS,
+                                                /* Say NumberOfPhysicalBreaks of 64 is required, it won't work
+                                                   if we optimize the user-input requirement to 65. So just
+                                                   let the user choose whether to use 64 or 33 and save him
+                                                   or her from unnecessary calculation */
+                                                adaptExt->scsi_config.max_sectors << SECTOR_SHIFT >> PAGE_SHIFT),
```

Comment:
To further elaborate, when a user *need* the clamp, the limit would most likely be a max_sectors queue limit, so if we don't do anything stupid, he or she can use the limit directly as the clamp (so no calculation needed).